Runtime/Scripts/RGBotConfigs/RGState.cs

```diff
@@ -28,6 +30,30 @@ public class RGState : MonoBehaviour
         public bool syncPosition = true;
         public bool syncRotation = true;
 
+        /*
+         * Add a method to our methodMap, with the given methodName id and delegate
+         * reference. Used to link state getter functions to their runtime delegates
+         */
+        protected void AddMethod(string methodName, Delegate methodDelegate)
+        {
+            methodMap[methodName] = methodDelegate;
+        }
+
+        /*
+         * Invokes a method with the given 'methodName' id with no parameters. Converts the
+         * return value to the given type and returns the value
+         */
+        protected object InvokeMethod(string methodName, Type returnType)
+        {
+            if (methodMap.TryGetValue(methodName, out var del))
+            {
+                return Convert.ChangeType(del.DynamicInvoke(), returnType);
+            }
+
+            RGDebug.LogError($"Method '{methodName}' not found");
+            return null;
+        }
```

Comment:
Is there a reason why we need to provide an Invoke pattern, vs just calling the function directly in the generated code?

Reply:
I can't see any downside - updated the generator to invoke directly